File: JavaScript/01.JavaScript-Basic-February-2021/09.whileLoop-Lab/09.moving.js
function moving(input) {
    let x = Number(input[0]);
    let y = Number(input[1]);
    let z = Number(input[2]);

    let availableSpace = x * y * z;
    let index = 3;

    while (input[index] != "Done"){
        availableSpace = availableSpace - Number(input[index]);
        if (availableSpace < 0) {
            console.log(`No more free space! You need ${Math.abs(availableSpace)} Cubic meters more.`);
            break;
        }
        index++;
    }

    if (availableSpace >= 0) {
        console.log(`${availableSpace} Cubic meters left.`);
    }
}

moving(["10", "10", "2", "20", "20", "20", "20", "122"]);
moving(["10", "1", "2", "4", "6", "Done"]);

File: Python/01.Python-Basic-April-2019/08.while_loop_lab/09.moving.py
length = int(input())
width = int(input())
height = int(input())

total_apartment_volume = length * width * height
total_boxes_volume = 0

is_free = True

boxes_count = input()

while not boxes_count == 'Done':
    boxes_count = int(boxes_count)
    total_boxes_volume += boxes_count

    if total_boxes_volume > total_apartment_volume:
        is_free = False
        break

    boxes_count = input()

if is_free:
    diff = total_apartment_volume - total_boxes_volume
    print(f'{diff} Cubic meters left.')
else:
    diff = total_boxes_volume - total_apartment_volume
    print(f'No more free space! You need {diff} Cubic meters more.')